packages/contracts-periphery/contracts/L1/messaging/L1ERC721Bridge.sol

```diff
@@ -1,168 +1,249 @@
         require(
-            deposits[_l1Token][_l2Token][_tokenId] == true,
+            deposits[_localToken][_remoteToken][_tokenId] == true,
             "Token ID is not escrowed in the L1 Bridge"
         );
 
-        deposits[_l1Token][_l2Token][_tokenId] = false;
+        deposits[_localToken][_remoteToken][_tokenId] = false;
 
         // When a withdrawal is finalized on L1, the L1 Bridge transfers the NFT to the withdrawer
         // slither-disable-next-line reentrancy-events
-        IERC721(_l1Token).transferFrom(address(this), _to, _tokenId);
+        IERC721(_localToken).transferFrom(address(this), _to, _tokenId);
+
+        // slither-disable-next-line reentrancy-events
+        emit ERC721BridgeFinalized(_localToken, _remoteToken, _from, _to, _tokenId, _data);
+    }
+
+    /**
+     * @notice Internal function for initiating a token bridge to the other domain.
+     *
+     * @param _localToken  Address of the ERC721 on this domain.
+     * @param _remoteToken Address of the ERC721 on the remote domain.
+     * @param _from        Address of the sender on this domain.
+     * @param _to          Address to receive the token on the other domain.
+     * @param _tokenId     Token ID to bridge.
+     * @param _minGasLimit Minimum gas limit for the bridge message on the other domain.
+     * @param _data        Optional data to forward to L2. Data supplied here will not be used to
+     *                     execute any code on L2 and is only emitted as extra data for the
+     *                     convenience of off-chain tooling.
+     */
+    function _initiateBridgeERC721(
+        address _localToken,
+        address _remoteToken,
+        address _from,
+        address _to,
+        uint256 _tokenId,
+        uint32 _minGasLimit,
+        bytes calldata _data
+    ) internal {
+        // When a deposit is initiated on L1, the L1 Bridge transfers the NFT to itself for future
+        // withdrawals.
+        // slither-disable-next-line reentrancy-events, reentrancy-benign
+        IERC721(_localToken).transferFrom(_from, address(this), _tokenId);
+
+        // Construct calldata for _l2Token.finalizeBridgeERC721(_to, _tokenId)
+        bytes memory message = abi.encodeWithSelector(
+            L2ERC721Bridge.finalizeBridgeERC721.selector,
+            _remoteToken,
+            _localToken,
+            _from,
+            _to,
+            _tokenId,
+            _data
+        );
+
+        // Send calldata into L2
+        // slither-disable-next-line reentrancy-events, reentrancy-benign
+        sendCrossDomainMessage(otherBridge, _minGasLimit, message);
+
+        // slither-disable-next-line reentrancy-benign
+        deposits[_localToken][_remoteToken][_tokenId] = true;
 
         // slither-disable-next-line reentrancy-events
-        emit ERC721WithdrawalFinalized(_l1Token, _l2Token, _from, _to, _tokenId, _data);
+        emit ERC721BridgeInitiated(_localToken, _remoteToken, _from, _to, _tokenId, _data);
```

Comment:
This ordering would be CEI compliant and probably allow us to remove some of the slither-disable comments:

```solidity
	// Construct calldata for _l2Token.finalizeBridgeERC721(_to, _tokenId)
        bytes memory message = abi.encodeWithSelector(
            L2ERC721Bridge.finalizeBridgeERC721.selector,
            _remoteToken,
            _localToken,
            _from,
            _to,
            _tokenId,
            _data
        );
        
        // When a deposit is initiated on L1, the L1 Bridge transfers the NFT to itself for future
        // withdrawals.
        deposits[_localToken][_remoteToken][_tokenId] = true;
        IERC721(_localToken).transferFrom(_from, address(this), _tokenId);

        // Send calldata into L2
        sendCrossDomainMessage(otherBridge, _minGasLimit, message);
        emit ERC721BridgeInitiated(_localToken, _remoteToken, _from, _to, _tokenId, _data);
    }
```